packages/liveblocks-react/src/comments/CommentsRoom.ts

```diff
@@ -80,7 +105,7 @@ function createOptimisticId(prefix: string) {
   return `${prefix}_${nanoid()}`;
 }
 
-export type RoomThreads<TThreadMetadata extends BaseMetadata> =
+export type State<Data = any> =
```

Comment:
1. The name `State` is very generic, even though this is a datastructure specifically used for threads only (due to the `threads` property in there). Maybe rename this to something more specific, or… do the opposite, and make this a truly generic type, using a `data` property here.
2. Try to avoid using default type params, i.e. avoid using `=` unless there is a good reason. Especially with generic types, you don't want to make it possible to use `const s = State;` without specifying the expected payload, for the same reason that `const arr = Array;` would not be valid.
3. Avoid using `any`s at all costs, especially as a library author. They infect, often unintentionally, many parts of our user's code bases for no good reason. Most of the time, just use `unknown` (or `never` if you don't anticipate using the key), but in this case, as per point (2), I would just remove it entirely.

Reply:
I started with wanting to build a generic cache manager but ended up pivoting to only threads. I agree the name is too generic. I've reverted the naming to use what we originally used:

```
export type RoomThreads<TThreadMetadata extends BaseMetadata> =
  | {
      isLoading: true;
      threads?: never;
      error?: never;
    }
  | {
      isLoading: false;
      threads?: never;
      error: Error;
    }
  | {
      isLoading: false;
      threads: ThreadData<TThreadMetadata>[];
      error?: never;
    };
```